src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -137,798 +34,233 @@ namespace JSONHelpers
-        bool res = true;
-        if (std::filesystem::exists(tmpFilePath))
+        for (const auto& [x, y, width, height] : canvasInfo.zones)
         {
-            auto deletedZoneSetsJson = json::from_file(tmpFilePath);
-            try
-            {
-                auto deletedCustomZoneSets = deletedZoneSetsJson->GetNamedArray(L"deleted-custom-zone-sets");
-                for (auto zoneSet : deletedCustomZoneSets)
-                {
-                    std::wstring uuid = L"{" + std::wstring{ zoneSet.GetString() } + L"}";
-                    customZoneSetsMap.erase(std::wstring{ uuid });
-                }
-            }
-            catch (const winrt::hresult_error&)
-            {
-                res = false;
-            }
-
-            DeleteTmpFile(tmpFilePath);
+            json::JsonObject zoneJson;
+            zoneJson.SetNamedValue(L"X", json::value(x));
+            zoneJson.SetNamedValue(L"Y", json::value(y));
+            zoneJson.SetNamedValue(L"width", json::value(width));
+            zoneJson.SetNamedValue(L"height", json::value(height));
+            zonesJson.Append(zoneJson);
         }
-
-        return res;
+        infoJson.SetNamedValue(L"zones", zonesJson);
+        return infoJson;
     }
 
-    bool FancyZonesData::ParseAppZoneHistory(const json::JsonObject& fancyZonesDataJSON)
+    std::optional<FancyZonesDataTypes::CanvasLayoutInfo> CanvasLayoutInfoJSON::FromJson(const json::JsonObject& infoJson)
     {
-        std::scoped_lock lock{ dataLock };
         try
         {
-            auto appLastZones = fancyZonesDataJSON.GetNamedArray(L"app-zone-history");
+            FancyZonesDataTypes::CanvasLayoutInfo info;
+            info.lastWorkAreaWidth = static_cast<int>(infoJson.GetNamedNumber(L"ref-width"));
+            info.lastWorkAreaHeight = static_cast<int>(infoJson.GetNamedNumber(L"ref-height"));
 
-            for (uint32_t i = 0; i < appLastZones.Size(); ++i)
+            json::JsonArray zonesJson = infoJson.GetNamedArray(L"zones");
+            uint32_t size = zonesJson.Size();
+            info.zones.reserve(size);
+            for (uint32_t i = 0; i < size; ++i)
             {
-                json::JsonObject appLastZone = appLastZones.GetObjectAt(i);
-                if (auto appZoneHistory = AppZoneHistoryJSON::FromJson(appLastZone); appZoneHistory.has_value())
-                {
-                    appZoneHistoryMap[appZoneHistory->appPath] = std::move(appZoneHistory->data);
-                }
-                else
-                {
-                    return false;
-                }
+                json::JsonObject zoneJson = zonesJson.GetObjectAt(i);
+                const int x = static_cast<int>(zoneJson.GetNamedNumber(L"X"));
+                const int y = static_cast<int>(zoneJson.GetNamedNumber(L"Y"));
+                const int width = static_cast<int>(zoneJson.GetNamedNumber(L"width"));
+                const int height = static_cast<int>(zoneJson.GetNamedNumber(L"height"));
```

Comment:
Replace all strings literals with const strings and explicitly mark them as non-localizable.